sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -117,6 +117,15 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
     resolve(name, children.flatMap(_.output), resolver, throwErrors)
 
   /**
+   * Optionally resolves the given string (`name`) to a [[NamedExpression]] using the input
+   * from all child nodes of this LogicalPlan. The given string is considered a string literal,
+   * i.e. the string itself should match the attribute name and the attribute name alone.
+   */
+  def resolveQuoted(name: String, resolver: Resolver): Option[NamedExpression] = {
+    output.find { attribute => resolver(name, attribute.name) }
```

Comment:
You can't always know if something is going to be ambiguous when you create the DataFrame as it might only be ambiguous at query time due to a setting like case sensitivity.